pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -131,6 +191,48 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Is the user banned? To determine this we query the BannedUser resource for any matching entries.  The query
+	// is based on the user's emailHash value - if there is a match, and the e-mail addresses are equal, then the
+	// user is banned.
+	banned := false
+
+	// Lookup the user email annotation
+	if emailLbl, exists := instance.Annotations[toolchainv1alpha1.UserSignupUserEmailAnnotationKey]; exists {
+
+		// Lookup the email hash label
+		if emailHashLbl, exists := instance.Labels[toolchainv1alpha1.UserSignupUserEmailHashAnnotationKey]; exists {
+
+			labels := map[string]string{toolchainv1alpha1.BannedUserEmailHashLabelKey: emailHashLbl}
+			opts := client.MatchingLabels(labels)
+			bannedUserList := &toolchainv1alpha1.BannedUserList{}
+
+			// Query BannedUser for resources that match the same email hash
+			if err = r.client.List(context.TODO(), bannedUserList, opts); err != nil {
+				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusFailedToReadBannedUsers, err, "Failed to query BannedUsers")
+			}
+
+			// One last check to confirm that the e-mail addresses match also (in case of the infinitesimal chance of a hash collision)
+			for _, bannedUser := range bannedUserList.Items {
+				if bannedUser.Spec.Email == emailLbl {
+					banned = true
+					break
+				}
+			}
+
+			hashIsValid, err := validateEmailHash(emailLbl, emailHashLbl)
+			if err != nil {
+				return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusFailedToValidateEmailHash)
```

Comment:
we should return the original error here:
```suggestion
				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, instance, r.setStatusFailedToValidateEmailHash, err, "Failed to validate email hash")
```

Reply:
If we return an error here, won't the controller then attempt to reconcile again ad infinitum?  Looking at the code, it seems that it is impossible for this error to actually ever manifest, and the only reason that the md5 digest's Write() function has an error in the return value is because it implements the `Writer` interface - the actual implementation of the method doesn't allow for an error to be returned.  Check it out for yourself - `err` is never assigned:

```
func (d *digest) Write(p []byte) (nn int, err error) {
	// Note that we currently call block or blockGeneric
	// directly (guarded using haveAsm) because this allows
	// escape analysis to see that p and d don't escape.
	nn = len(p)
	d.len += uint64(nn)
	if d.nx > 0 {
		n := copy(d.x[d.nx:], p)
		d.nx += n
		if d.nx == BlockSize {
			if haveAsm {
				block(d, d.x[:])
			} else {
				blockGeneric(d, d.x[:])
			}
			d.nx = 0
		}
		p = p[n:]
	}
	if len(p) >= BlockSize {
		n := len(p) &^ (BlockSize - 1)
		if haveAsm {
			block(d, p[:n])
		} else {
			blockGeneric(d, p[:n])
		}
		p = p[n:]
	}
	if len(p) > 0 {
		d.nx = copy(d.x[:], p)
	}
	return
}
```

Reply:
ok I see...
then lets just ignore  the error (with a comment!) in `validateEmailHash()` as you did previously:
```
func validateEmailHash(...) bool {
   ...
    _, _ := d.write(...)
   ...
}
```

Reply:
Fixed